### DISCUSSION TITLE
Close Drawer window with button

### DISCUSSION THREAD

**Discussion by AcernisSupport** [ant-design-blazor/ant-design-blazor#3551] [Q&A]
Good afternoon list
I am using the Drawer control to be able to insert/modify the data of a DTO and then with a button I save them in the db.
Up to here everything works.
My problem is that once I use the SAVE button I would like the control to close.
I tried to see but I can't figure out how to do it.
If anyone has a simple example so I can understand
I call the control with this a button which onClick makes this call

`public void OnAddAttribute01()
{
ShowAttribute01 = true;
}`

and this is the control in the razor part

`<Drawer Id="Attribure01" Width="500" Visible=ShowAttribiute01 OnClose="@(() => ShowAttribiute01 = false)" Title='("Insertion/Modification")'>
          <Button Style="justify-content:center" Type="@ButtonType.Primary" OnClick="SaveAddAttribute01">@Locale["Common.Save"]</Button>
</Drawer>`

Thanks and good job

**Comment by ElderJames** [comment]
cc @zxyao145

**Reply by zxyao145** [reply]
@AcernisSupport Hi, please try the following code:

```razor

<Button Type="@ButtonType.Primary" OnClick="@(() => ShowAttribiute01 = true)">
    Open Drawer
</Button>

<Drawer Id="Attribure01" Width="500"
        Visible=ShowAttribiute01
        OnClose="@(() => ShowAttribiute01 = false)"
        Title='("Insertion/Modification")'>

    <Button Style="justify-content:center" Type="@ButtonType.Primary" OnClick="SaveAddAttribute01">
        Save
    </Button>
</Drawer>
@code {

    private bool ShowAttribiute01 = false;

    private async Task SaveAddAttribute01()
    {
        // await save to db
        Console.WriteLine("Save to db");
+        ShowAttribiute01 = false;
    }
}
```

![2023-12-03_23-12-47](https://github.com/ant-design-blazor/ant-design-blazor/assets/24973084/cde2f2ec-9dc4-4fa1-a8e9-8765735959c0)

**Comment by AcernisSupport** [comment]
Good afternoon

Thanks to zxyao145 and also colleague Elderjames

It works perfectly

Again good work everyone
